h2o-persist-http/src/main/java/water/persist/PersistHTTP.java

```diff
@@ -93,35 +100,49 @@ private static long parseContentRangeLength(Header contentRange) {
     String[] range = crParts[0].split("-");
     if (range.length != 2)
       throw new IllegalStateException("Invalid HTTP response. Cannot interpret range value in response header " + HttpHeaders.CONTENT_RANGE + ": " + contentRange.getValue());
-    return 1 + Long.valueOf(range[1]) - Long.valueOf(range[0]);
+    return 1 + Long.parseLong(range[1]) - Long.parseLong(range[0]);
   }
 
   private static URI decodeKey(Key k) {
     return URI.create(new String((k._kb[0] == Key.CHK) ? Arrays.copyOfRange(k._kb, Vec.KEY_PREFIX_LEN, k._kb.length) : k._kb));
   }
 
+  long useLazyLoad(URI uri) throws IOException {
+    HttpRequestBase req = createReq(uri, true);
+    try (CloseableHttpClient client = HttpClientBuilder.create().build();
+         CloseableHttpResponse response = client.execute(req)) {
+      if (isCompressed(response)) 
+        return -1L; // avoid lazy-loading of compressed resource that cannot be parsed in parallel
+      return checkRangeSupport(uri, response);
+    }
+  }
+
+  static boolean isCompressed(HttpResponse response) {
+    Header contentTypeHeader = response.getFirstHeader(HttpHeaders.CONTENT_TYPE);
+    if (contentTypeHeader == null)
+      return false; // assume not compressed
+    String contentType = contentTypeHeader.getValue();
+    return COMPRESSED_CONTENT_TYPES.contains(contentType);
```

Comment:
I am afraid content type can be case insensitive so contentType.toLowerCase() to play it safe?